src/AggregateFunctions/AggregateFunctionSparkbar.h

```diff
@@ -0,0 +1,303 @@
+#pragma once
+
+#include <DataTypes/DataTypeString.h>
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <common/range.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Columns/ColumnString.h>
+#include <common/logger_useful.h>
+#include <IO/ReadBufferFromString.h>
+
+namespace DB
+{
+
+template<typename X, typename Y>
+struct AggregateFunctionSparkbarData
+{
+    using Points = std::map<X, Y>;
+
+    Points points;
+
+    Y min_y = std::numeric_limits<Y>::max();
+    Y max_y = std::numeric_limits<Y>::min();
+
+    void add(X x, Y y)
+    {
+        auto it = points.find(x);
+        if (it != points.end())
+            it->second += y;
+        else
+            points.emplace(x, y);
+        if (y < min_y)
+            min_y = y;
+        if (y > max_y)
+            max_y = y;
+    }
+
+    void merge(const AggregateFunctionSparkbarData & other)
+    {
+        if (other.points.empty())
+            return;
+
+        for (auto & point : other.points)
+        {
+            auto it = points.find(point.first);
+            if (it != points.end())
+                it->second += point.second;
+            else
+                points.emplace(point.first, point.second);
+        }
+
+        if (other.min_y < min_y)
+            min_y = other.min_y;
+
+        if (other.max_y > max_y)
+            max_y = other.max_y;
+    }
+
+    void serialize(WriteBuffer & buf) const
+    {
+        writeBinary(min_y, buf);
+        writeBinary(max_y, buf);
+        writeVarUInt(points.size(), buf);
+
+        for (const auto & elem : points)
+        {
+            writeBinary(elem.first, buf);
+            writeBinary(elem.second, buf);
+        }
+    }
+
+    void deserialize(ReadBuffer & buf)
+    {
+        readPODBinary(min_y, buf);
+        readPODBinary(max_y, buf);
+        size_t size;
+        readVarUInt(size, buf);
+
+        /// TODO Protection against huge size
+
+        points.clear();
+
+        X x;
+        Y y;
+        for (size_t i = 0; i < size; ++i)
+        {
+            readBinary(x, buf);
+            readBinary(y, buf);
+            points.emplace(x, y);
+        }
+    }
+
+};
+
+template<typename X, typename Y>
+class AggregateFunctionSparkbar final
+    : public IAggregateFunctionDataHelper<AggregateFunctionSparkbarData<X, Y>, AggregateFunctionSparkbar<X, Y>>
+{
+
+private:
+    size_t width;
+    X min_x;
+    X max_x;
+
+    String getBar(const UInt8 value) const
+    {
+        // ▁▂▃▄▅▆▇█
```

Comment:
Is it possible to make these bars aligned to the bottom line?

Reply:
Is it aligned with the bottom line of ```└─────┘``` ​​output by clickhouse-client?

Reply:
![image](https://user-images.githubusercontent.com/5085485/128863863-0ac9a2c1-51ba-40f5-b7bf-7a90a3eebba1.png)

Ohh, it's rendered incorrectly in my browser. It looks perfectly in Android)

Reply:
Yes, it depends on font.

Liberation Mono - bad;
DejaVu Sans Mono - good.

Also some terminal emulators may have their own rendering of block rendering characters, independent on font.
Check my pet-project if you wonder: https://alexey-milovidov.github.io/font-selector/